core/src/main/java/org/elasticsearch/index/query/morelikethis/Item.java

```diff
@@ -0,0 +1,339 @@
+     */
+    public Item doc(BytesReference doc) {
+        this.doc = doc;
+        return this;
+    }
+
+    /**
+     * Sets to a given artificial document, that is a document that is not present in the index.
+     */
+    public Item doc(XContentBuilder doc) {
+        return this.doc(doc.bytes());
+    }
+
+    public String[] fields() {
+        return fields;
+    }
+
+    public Item fields(String... fields) {
+        this.fields = fields;
+        return this;
+    }
+
+    public Map<String, String> perFieldAnalyzer() {
+        return perFieldAnalyzer;
+    }
+
+    /**
+     * Sets the analyzer(s) to use at any given field.
+     */
+    public Item perFieldAnalyzer(Map<String, String> perFieldAnalyzer) {
+        this.perFieldAnalyzer = perFieldAnalyzer;
+        return this;
+    }
+
+    public String routing() {
+        return routing;
+    }
+
+    public Item routing(String routing) {
+        this.routing = routing;
+        return this;
+    }
+
+    public long version() {
+        return version;
+    }
+
+    public Item version(long version) {
+        this.version = version;
+        return this;
+    }
+
+    public VersionType versionType() {
+        return versionType;
+    }
+
+    public Item versionType(VersionType versionType) {
+        this.versionType = versionType;
+        return this;
+    }
+
+    /**
+     * Convert this to a {@link TermVectorsRequest} for fetching the terms of the document.
+     */
+    public TermVectorsRequest toTermVectorsRequest() {
+        TermVectorsRequest termVectorsRequest = new TermVectorsRequest(index, type, id)
+                .selectedFields(fields)
+                .routing(routing)
+                .version(version)
+                .versionType(versionType)
+                .perFieldAnalyzer(perFieldAnalyzer)
+                .positions(false)  // ensures these following parameters are never set
+                .offsets(false)
+                .payloads(false)
+                .fieldStatistics(false)
+                .termStatistics(false)
+                .dfs(false);
+        // for artificial docs to make sure that the id has changed in the item too
+        if (doc != null) {
+            termVectorsRequest.doc(doc, true);
+            this.id(termVectorsRequest.id());
+        }
+        return termVectorsRequest;
+    }
+
+    /**
+     * Parses and returns the given item.
+     */
+    public static Item parse(XContentParser parser, Item item) throws IOException {
+        XContentParser.Token token;
+        String currentFieldName = null;
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            } else if (currentFieldName != null) {
+                if (ParseFieldMatcher.STRICT.match(currentFieldName, Field.INDEX)) {
```

Comment:
I think it is. For instance if some day I want to deprecate one option, I will just modify the ParseField and think I'm done. But it won't because the ParseFieldMatcher is hardcoded here?